src/lib/prover/prover.ml

```diff
@@ -318,6 +318,29 @@ let create ~logger ~pids ~conf_dir ~proof_level ~constraint_constants =
       ] ;
   Child_processes.Termination.register_process pids process
     Child_processes.Termination.Prover ;
+  let exit_or_signal =
+    Child_processes.Termination.wait_safe ~logger process ~module_:__MODULE__
+      ~location:__LOC__ ~here:[%here]
+  in
+  don't_wait_for
+    ( match%map exit_or_signal with
+    | Ok (Ok ()) ->
+        [%log fatal] "Unexpected prover termination" ;
+        failwith "Unexpected prover termination"
```

Comment:
Won't these be caught by the monitor, rather than bubbling up and exiting the process?